File: samples/CameraAccessAndroid/app/src/main/java/com/meta/wearable/dat/externalsampleapps/cameraaccess/conference/ConferenceContact.kt
package com.meta.wearable.dat.externalsampleapps.cameraaccess.conference

enum class ConferenceEnrichmentStatus(val wireValue: String) {
    NOT_REQUESTED("notRequested"),
    QUEUED("queued"),
    ENRICHING("enriching"),
    COMPLETED("completed"),
    FAILED("failed");

    val displayName: String
        get() = when (this) {
            NOT_REQUESTED -> "Not Requested"
            QUEUED -> "Queued"
            ENRICHING -> "Enriching"
            COMPLETED -> "Enriched"
            FAILED -> "Needs Retry"
        }

    companion object {
        fun fromValue(value: String?): ConferenceEnrichmentStatus {
            return entries.firstOrNull { it.wireValue == value } ?: NOT_REQUESTED
        }
    }
}

data class ConferenceEnrichmentPayload(
    val headline: String? = null,
    val companySummary: String? = null,
    val talkingPoints: List<String> = emptyList(),
    val followUp: String? = null,
    val confidenceNotes: String? = null,
    val sourceUrls: List<String> = emptyList(),
    val rawResponse: String? = null,
) {
    val summaryText: String
        get() = listOfNotNull(headline, companySummary, followUp)
            .map { it.trim() }
            .filter { it.isNotEmpty() }
            .joinToString(separator = "\n\n")
}

data class ConferenceContact(
    val id: String,
    val dedupeKey: String,
    val name: String,
    val company: String?,
    val role: String?,
    val sourceType: ConferenceSourceType,
    val confidence: Double,
    val observedText: String?,
    val disposition: ConferenceExtractionDisposition,
    val firstSeenAtMs: Long,
    val lastSeenAtMs: Long,
    val enrichmentStatus: ConferenceEnrichmentStatus,
    val enrichment: ConferenceEnrichmentPayload?,
    val enrichmentError: String?,
    val lastEnrichedAtMs: Long?,
    val conversationSnippet: String? = null,
    val lastConversationAtMs: Long? = null,
) {
    val summaryText: String
        get() = listOfNotNull(company?.trim()?.takeIf { it.isNotEmpty() }, role?.trim()?.takeIf { it.isNotEmpty() })
            .joinToString(separator = " / ")

    val canRetryEnrichment: Boolean
        get() = disposition == ConferenceExtractionDisposition.ACCEPTED &&
            enrichmentStatus == ConferenceEnrichmentStatus.FAILED

    val isEnrichmentInFlight: Boolean
        get() = enrichmentStatus == ConferenceEnrichmentStatus.QUEUED ||
            enrichmentStatus == ConferenceEnrichmentStatus.ENRICHING

    val hasConversationSnippet: Boolean
        get() = !conversationSnippet.isNullOrBlank()

    val lastActivityAtMs: Long
        get() = lastConversationAtMs ?: lastSeenAtMs
}

File: samples/CameraAccess/CameraAccess/iPhone/ConferenceMode/ConferenceContact.swift
import Foundation

enum ConferenceEnrichmentStatus: String, CaseIterable, Codable, Equatable {
  case notRequested
  case queued
  case enriching
  case completed
  case failed

  var displayName: String {
    switch self {
    case .notRequested: return "Not Requested"
    case .queued: return "Queued"
    case .enriching: return "Enriching"
    case .completed: return "Enriched"
    case .failed: return "Needs Retry"
    }
  }
}

struct ConferenceEnrichmentPayload: Codable, Equatable {
  let headline: String?
  let companySummary: String?
  let talkingPoints: [String]
  let followUp: String?
  let confidenceNotes: String?
  let sourceURLs: [String]
  let rawResponse: String?

  var summaryText: String {
    [headline, companySummary, followUp]
      .compactMap { value in
        guard let trimmed = value?.trimmingCharacters(in: .whitespacesAndNewlines),
              !trimmed.isEmpty else { return nil }
        return trimmed
      }
      .joined(separator: "\n\n")
  }
}

struct ConferenceContact: Identifiable, Equatable {
  let id: String
  let dedupeKey: String
  let name: String
  let company: String?
  let role: String?
  let sourceType: ConferenceSourceType
  let confidence: Double
  let observedText: String?
  let disposition: ConferenceExtractionDisposition
  let firstSeenAt: Date
  let lastSeenAt: Date
  let enrichmentStatus: ConferenceEnrichmentStatus
  let enrichment: ConferenceEnrichmentPayload?
  let enrichmentError: String?
  let lastEnrichedAt: Date?

  var summaryText: String {
    [company, role]
      .compactMap { value in
        guard let trimmed = value?.trimmingCharacters(in: .whitespacesAndNewlines),
              !trimmed.isEmpty else { return nil }
        return trimmed
      }
      .joined(separator: " / ")
  }
}
